File: DSAlgorithms/Leetcode/c++/arrays_hash/392-is_subsequence.cpp
bool is_subsequence(string s, string t) 
{
    int first = 0;
    for(int i = 0; i < t.size(); i++) 
    {
        if(s[first] == t[i])
            first++;
    }
    if(first < s.size())
        return false;

    return true;
}


bool is_subsequence(string s, string t) 
{
    int i = 0;
    int j = 0;
    while(i < s.size() && j < t.size())
    {
        if(s[i] == t[j])
            i++;
        j++;
    }
    if(i == s.size())
        return true;
    return false;
}

File: DSAlgorithms/Leetcode/ts/src/arrays_hash/392-is_subsequence.ts
const isSubsequence = (s: string, t: string): boolean => {
   let first: number = 0;
   for(let current = 0; current < t.length; current++) 
   {
       if(s[first] === t[current])
            first++;
   }
   if(first >= s.length)
        return true;
    return false;
};
